test/lint/lint-circular-dependencies.sh

```diff
@@ -100,6 +100,7 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "evo/chainhelper -> masternode/payments -> validation -> evo/chainhelper"
     "evo/deterministicmns -> validationinterface -> evo/deterministicmns"
     "logging -> util/system -> sync -> logging/timer -> logging"
+    "rpc/blockchain -> rpc/server -> rpc/blockchain"
```

Comment:
let's move `EnsureAnyNodeContext` to other header to avoid this circular dependency

Reply:
`EnsureAnyNodeContext` originates from upstream so I'm not comfortable moving it to an entirely different file.

Reply:
can you consider to backport bitcoin/bitcoin#20295 then?

Reply:
20295 backport should be done in a separate PR imo